### PR TITLE
Allow quiet error logs option to be set from next.config.js

### PR DESCRIPTION
**Pull request by kconner**
This PR adds an option to next.config.js, `experimental.quietErrorLog`, which has the same effect as `quiet: true`.

Background:

There have been some issues raised related to custom logging, notably #4808. A basic part of the need is to disable built-in logging of thrown errors, so you can get to a clean slate and then add your own logs.

That feature exists: You can pass the `quiet: true` option when creating the server. However you must build a custom server implementation to do that. I'd rather avoid it, because staying current with a stock configuration pays benefits, and I'm happy with Next's standard routing. Furthermore, the [custom server documentation page](https://nextjs.org/docs/advanced-features/custom-server) indicates that building a custom server will sacrifice some standard features:

>Before deciding to use a custom server please keep in mind that it should only be used when the integrated router of Next.js can't meet your app requirements. A custom server will remove important performance optimizations…

In summary, I want to use the `quiet: true` option without making a custom server, and next.config.js seems like a suitable place to expose the option.

I'm happy to do additional legwork if this feature is wanted. Thank you!

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| buildDuration | 12.2s | 12.3s | ⚠️ +115ms |
| nodeModulesSize | 43.6 MB | 43.6 MB | ⚠️ +239 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.065 | 2.051 | -0.01 |
| / avg req/sec | 1210.72 | 1218.81 | +8.09 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.251 | 1.272 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 1999.07 | 1965.35 | ⚠️ -33.72 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| 677f882d2ed8..a2e7.js gzip | 13.4 kB | 13.4 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 7.14 kB | 7.14 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 60.3 kB | 60.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| _app-fde3324..9dd1.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-af59f..582f.js gzip | 3.46 kB | 3.46 kB | ✓ |
| amp-9716187d..0aa8.js gzip | 536 B | 536 B | ✓ |
| hooks-107e90..74c7.js gzip | 888 B | 888 B | ✓ |
| index-ac435c..ecf2.js gzip | 227 B | 227 B | ✓ |
| link-e6cee42..49eb.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..dc9d.js gzip | 303 B | 303 B | ✓ |
| withRouter-6..0e02.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.64 kB | 8.64 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| _buildManifest.js gzip | 346 B | 346 B | ✓ |
| Overall change | 346 B | 346 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| index.html gzip | 612 B | 612 B | ✓ |
| link.html gzip | 621 B | 621 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| buildDuration | 14s | 14.2s | ⚠️ +175ms |
| nodeModulesSize | 43.6 MB | 43.6 MB | ⚠️ +239 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| 677f882d2ed8..a2e7.js gzip | 13.4 kB | 13.4 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 7.14 kB | 7.14 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 60.3 kB | 60.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| _app-fde3324..9dd1.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-af59f..582f.js gzip | 3.46 kB | 3.46 kB | ✓ |
| amp-9716187d..0aa8.js gzip | 536 B | 536 B | ✓ |
| hooks-107e90..74c7.js gzip | 888 B | 888 B | ✓ |
| index-ac435c..ecf2.js gzip | 227 B | 227 B | ✓ |
| link-e6cee42..49eb.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..dc9d.js gzip | 303 B | 303 B | ✓ |
| withRouter-6..0e02.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.64 kB | 8.64 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| _buildManifest.js gzip | 346 B | 346 B | ✓ |
| Overall change | 346 B | 346 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| _error.js | 1.02 MB | 1.02 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| 500.html | 2.65 kB | 2.65 kB | ✓ |
| amp.amp.html | 10.6 kB | 10.6 kB | ✓ |
| amp.html | 1.86 kB | 1.86 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1.02 MB | 1.02 MB | ✓ |
| link.js | 1.08 MB | 1.08 MB | ✓ |
| routerDirect.js | 1.07 MB | 1.07 MB | ✓ |
| withRouter.js | 1.07 MB | 1.07 MB | ✓ |
| Overall change | 5.27 MB | 5.27 MB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 5 Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| buildDuration | 14s | 14.2s | ⚠️ +203ms |
| nodeModulesSize | 43.6 MB | 43.6 MB | ⚠️ +239 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.021 | 2.043 | ⚠️ +0.02 |
| / avg req/sec | 1237.07 | 1223.44 | ⚠️ -13.63 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.225 | 1.29 | ⚠️ +0.06 |
| /error-in-render avg req/sec | 2041.5 | 1938.19 | ⚠️ -103.31 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| 597-12fee761..569d.js gzip | 13.3 kB | 13.3 kB | ✓ |
| framework.HASH.js gzip | 39.3 kB | 39.3 kB | ✓ |
| main-HASH.js gzip | 7.11 kB | 7.11 kB | ✓ |
| webpack-HASH.js gzip | 993 B | 993 B | ✓ |
| Overall change | 60.7 kB | 60.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| _app-b5edc60..e5c2.js gzip | 1.29 kB | 1.29 kB | ✓ |
| _error-b58c1..9b8e.js gzip | 3.4 kB | 3.4 kB | ✓ |
| amp-89a5460c..567f.js gzip | 558 B | 558 B | ✓ |
| hooks-8c2e74..be37.js gzip | 924 B | 924 B | ✓ |
| index-fec729..83b2.js gzip | 243 B | 243 B | ✓ |
| link-3631090..4cab.js gzip | 1.65 kB | 1.65 kB | ✓ |
| routerDirect..20d7.js gzip | 328 B | 328 B | ✓ |
| withRouter-b..a4b9.js gzip | 326 B | 326 B | ✓ |
| Overall change | 8.72 kB | 8.72 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| _buildManifest.js gzip | 325 B | 325 B | ✓ |
| Overall change | 325 B | 325 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | kconner/next.js quiet-config-option | Change |
| - | - | - | - |
| index.html gzip | 585 B | 585 B | ✓ |
| link.html gzip | 594 B | 594 B | ✓ |
| withRouter.html gzip | 582 B | 582 B | ✓ |
| Overall change | 1.76 kB | 1.76 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/",
         "query": {},
-        "buildId": "2BBTcaKdTBB7ftaWYgSHq",
+        "buildId": "5oSa0lJzIFse5kGPizwKn",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/2BBTcaKdTBB7ftaWYgSHq/_buildManifest.js"
+      src="/_next/static/5oSa0lJzIFse5kGPizwKn/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/2BBTcaKdTBB7ftaWYgSHq/_ssgManifest.js"
+      src="/_next/static/5oSa0lJzIFse5kGPizwKn/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -48,7 +48,7 @@
         "props": { "pageProps": {} },
         "page": "/link",
         "query": {},
-        "buildId": "2BBTcaKdTBB7ftaWYgSHq",
+        "buildId": "5oSa0lJzIFse5kGPizwKn",
         "isFallback": false,
         "gip": true
       }
@@ -82,11 +82,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/2BBTcaKdTBB7ftaWYgSHq/_buildManifest.js"
+      src="/_next/static/5oSa0lJzIFse5kGPizwKn/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/2BBTcaKdTBB7ftaWYgSHq/_ssgManifest.js"
+      src="/_next/static/5oSa0lJzIFse5kGPizwKn/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/withRouter",
         "query": {},
-        "buildId": "2BBTcaKdTBB7ftaWYgSHq",
+        "buildId": "5oSa0lJzIFse5kGPizwKn",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/2BBTcaKdTBB7ftaWYgSHq/_buildManifest.js"
+      src="/_next/static/5oSa0lJzIFse5kGPizwKn/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/2BBTcaKdTBB7ftaWYgSHq/_ssgManifest.js"
+      src="/_next/static/5oSa0lJzIFse5kGPizwKn/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
</details>
Commit: a7090de6a54b88286054dcdbd5e0b5631ef38b0c

**Comment by kconner** [comment]
We worked around this by no longer using `_error`. Instead, our application component presents our error page body component, instead of the page route component, when certain server side props are produced from any page route. We produce those by specifically catching errors in `getServerSideProps` and not letting them fall through to Next, if we can help it. We think this effectively replicates Next's _error presentation behavior without its default logging. It was also desirable for getting server side dynamic error presentation without adding to the browser JS bundle – getInitialProps was leaky in that respect since it can also execute in the browser.
